pyuvsim/tests/test_simsetup.py

```diff
@@ -433,11 +473,76 @@ def test_param_select_redundant():
+                                        telescope_location=(-30.72152777777791, 21.428305555555557, 1073.0000000093132),
+                                        telescope_name="HERA", Nfreqs=10, start_freq=1e8, bandwidth=1e8, Ntimes=60,
+                                        integration_time=100.0, start_time=2458101.0, polarizations=['xx'], bls=bls,
+                                        fill_blts=True, run_check=True)
+    nt.assert_equal(uvd.Nbls, len(bls))
+    # also check that '1' gets converted to [1]
+    uvd = pyuvsim.simsetup.setup_uvdata(array_layout=os.path.join(SIM_DATA_PATH, "test_config/triangle_bl_layout.csv"),
+                                        telescope_location=(-30.72152777777791, 21.428305555555557, 1073.0000000093132),
+                                        telescope_name="HERA", Nfreqs=10, start_freq=1e8, bandwidth=1e8, Ntimes=60,
+                                        integration_time=100.0, start_time=2458101.0, polarizations=['xx', 'yy'], bls=bls,
+                                        fill_blts=True, antenna_nums='1',
+                                        no_autos=False, run_check=True)
+    nt.assert_equal(uvd.Nbls, 1)
+    nt.assert_equal(uvd.Npols, 2)
+
+    # check time and freq array definitions supersede other parameters
+    fa = np.linspace(100, 200, 11)
+    ta = np.linspace(2458101, 2458102, 21)
+    uvd = pyuvsim.simsetup.setup_uvdata(array_layout=os.path.join(SIM_DATA_PATH, "test_config/triangle_bl_layout.csv"),
+                                        telescope_location=(-30.72152777777791, 21.428305555555557, 1073.0000000093132),
+                                        telescope_name="HERA", Nfreqs=10, start_freq=1e8, bandwidth=1e8, freq_array=fa,
+                                        Ntimes=60, integration_time=100.0, start_time=2458101.0, time_array=ta,
+                                        polarizations=['xx'], no_autos=True, fill_blts=True, run_check=True)
+    nt.assert_equal(uvd.Ntimes, 21)
+    nt.assert_equal(uvd.Nfreqs, 11)
+
+    # test incomplete uvdata setup
+    uvd = pyuvsim.simsetup.setup_uvdata(array_layout=os.path.join(SIM_DATA_PATH, "test_config/triangle_bl_layout.csv"),
+                                        telescope_location=(-30.72152777777791, 21.428305555555557, 1073.0000000093132),
+                                        telescope_name="HERA", Nfreqs=10, start_freq=1e8, bandwidth=1e8, Ntimes=60,
+                                        integration_time=100.0, start_time=2458101.0, polarizations=['xx'], no_autos=True,
+                                        fill_blts=False, run_check=True)
+    attrs = ['data_array', 'nsample_array', 'flag_array', 'integration_time', 'ant_1_array', 'ant_2_array']
+    for attr in attrs:
+        nt.assert_true(getattr(uvd, attr) is None)
+
+    # test feeding array layout as dictionary
+    antpos, ants = uvd.get_ENU_antpos()
+    antpos_d = dict(zip(ants, antpos))
```

Comment:
There shouldn't be two forms for the `antenna_layout` can be passed to `parse_telescope_params`. When read from a csv file, it becomes a record array with keys for antenna name, number, beam_ID, E, N, and U. When passed as a dictionary, the only information carried are antenna numbers and positions. Maybe `antenna_layout_csv` is a misleading name since it specifies more than just the layout.